### ISSUE TITLE
installation of module failed 

### ISSUE THREAD

**Issue by betsyrosemanuel** [CESNET/libyang#305]
Hi,
I was unable to install the module "bbf-interface-port-reference" and got the following error

No resolvents found for leafref "/ietf-entity:entity/ietf-entity:physical-entity/ietf-entity:name".
[ERR] (srctl_ly_log_cb:479) libyang: Leafref is not conditional based on "entity-config" feature as its target.
[ERR] (srctl_ly_log_cb:479) libyang: Module "bbf-interface-port-reference" parsing failed.
Error: Unable to load the module by libyang.
Install operation failed.

The module is as follows 

```
module bbf-interface-port-reference {

  namespace "urn:broadband-forum-org:bbf-interface-port-reference";
  
  prefix bbf-if-port-ext;
  import iana-if-type {
    prefix ianaift;
  }
  import ietf-interfaces {
    prefix if;
  }
  import ietf-entity {
    prefix ent;
  }
  import iana-entity {
    prefix ianaent;
  }
  import bbf-entity-extension {
    prefix bbf-ent-ext;
  }

  organization

  revision 2016-07-07 {
    //
  }
  
  typedef entity-ref {
    type leafref {
      path "/ent:entity/ent:physical-entity/ent:name";
    }
    description
      "Used to reference to a port entity";
  }

  augment "/if:interfaces/if:interface" {
   // when "if:type = 'ianaift:fastdsl' or if:type ='ianaift:ethernetCsmacd'"; 
    
    leaf port-layer-if {
      type entity-ref;
      must "/ent:entity/ent:physical-entity[current()]/bbf-ent-ext:class = 'ianaent:port'" {
        error-message "Referenced entity is not of class port identitiy";
      }
  //    mandatory true;    
      description
        "A references to the physical port entity layered on top of this interface.";
    }
  }
```

**Comment by michalvasko** [comment]
Hi,
the error is correct and the models should be fixed. Testing this in *yanglint*, it prints also the path for every error, so the full output is:
```
libyang[0]: No resolvents found for leafref "/ietf-entity:entity/ietf-entity:physical-entity/ietf-entity:name". (path: /ietf-interfaces:interfaces/ietf-interfaces:interface/bbf-interface-port-reference:port-layer-if)
libyang[0]: Leafref is not conditional based on "entity-config" feature as its target. (path: /ietf-interfaces:interfaces/ietf-interfaces:interface/bbf-interface-port-reference:port-layer-if)
libyang[0]: Module "bbf-interface-port-reference" parsing failed.
```

The error is caused, in short, by not conforming to [RFC 7950 section 9.9](https://tools.ietf.org/html/rfc7950#section-9.9), last paragraph.

Regards,
Michal

**Comment by betsyrosemanuel** [comment]
hi,
 I have used the same module and performed <edit-config> for the xml in the file attached using the CONFD and got a <rpc-reply> too. But i am unable to install the same in sysrepo.. If it was a problem in the module itself as you said then I should have gotten an error-reply when i performed the operation in CONFD. 
 
also i got the same error as follows while installing the module "bbf-entity-extension" initially.
[ERR] (srctl_ly_log_cb:479) libyang: Leafref is not conditional based on "entity-config" feature as its target.
[ERR] (srctl_ly_log_cb:479) libyang: Module "bbf-interface-port-reference" parsing failed.
Error: Unable to load the module by libyang.
Install operation failed. 
 
Using the new updated devel branch I was finally able to install it.

**Comment by michalvasko** [comment]
Hi,
I got lost a bit, so does it work for you now or not? What I said before should be correct, the model really is wrong and CONFD probably does not check this particular condition, it does not mean anything else when it does not complain. If you turn the feature `entity-config` off, it will likely encounter a problem.

Regards,
Michal

**Comment by betsyrosemanuel** [comment]
Hi,
     irrespective of whether the feature is enabled or disabled I still get the same error.
What I was actually trying to convey is that I faced the same issue almost a month ago while I tried to install the module 'bbf-entity-extension' (using old master branch). Now I used the new devel branch and I was able to install 'bbf-entity-extension' successfully. I have encountered the same problem now while trying to install the module 'bbf-interface-port-reference'.
